katapult_test.go

```diff
@@ -785,27 +787,38 @@ func TestNew(t *testing.T) {
 func TestWithHTTPClient(t *testing.T) {
 	hc := &http.Client{Timeout: 42 * time.Second}
 	c := &Client{HTTPClient: http.DefaultClient}
-	err := WithHTTPClient(hc)(c)
+	err := WithHTTPClient(hc)(c, nil)
 	assert.NoError(t, err)
 	assert.Equal(t, hc, c.HTTPClient)
 }
 
 func TestWithUserAgent(t *testing.T) {
 	c := &Client{}
 	ua := "roger_moore/0.0.7"
-	err := WithUserAgent(ua)(c)
+	err := WithUserAgent(ua)(c, nil)
 	assert.NoError(t, err)
 	assert.Equal(t, ua, c.UserAgent)
 }
 
 func TestWithAPIKey(t *testing.T) {
 	c := &Client{}
 	key := "extremely_very_secret_secret"
-	err := WithAPIKey(key)(c)
+	err := WithAPIKey(key)(c, nil)
 	assert.NoError(t, err)
 	assert.Equal(t, key, c.APIKey)
 }
 
+func TestWithTracing(t *testing.T) {
+	c := &Client{}
+	httpClient := &http.Client{}
+	c.HTTPClient = httpClient
+
+	err := WithTracing()(c, httpClient)
+	assert.NoError(t, err)
+
+	assert.IsType(t, otelhttp.NewTransport(nil), httpClient.Transport)
+}
```

Comment:
This currently does not guarantee that `WithTracing()` only modifies the given client, rather than `c.HTTPClient`.

So I think we'll need two `http.Client` instances, one on `c` itself, and the existing one we have, and then assert that the transport was modified on the given client, but not on `c.HTTPClient`.

Also this options kind of feels like it should be tested as part of a the `New()` function, as its behavior depends a bit on logic within that function, and also if a `WithHTTPClient()` option was provided or not, as it technically has zero effect on the final `katapult.Client` when both options are used.